Lib/locale.py

```diff
@@ -91,6 +91,26 @@ def setlocale(category, value=None):
             raise Error('_locale emulation only supports "C" locale')
         return 'C'
 
+else:
+    _conditional_constants_names = ["ABDAY_1", "ABDAY_2", "ABDAY_3",
+                                    "ABDAY_4", "ABDAY_5", "ABDAY_6",
+                                    "ABDAY_7", "ABMON_1", "ABMON_10",
+                                    "ABMON_11", "ABMON_12", "ABMON_2",
+                                    "ABMON_3", "ABMON_4", "ABMON_5",
+                                    "ABMON_6", "ABMON_7", "ABMON_8",
+                                    "ABMON_9", "ALT_DIGITS", "CODESET",
+                                    "CRNCYSTR", "DAY_1", "DAY_2", "DAY_3",
+                                    "DAY_4", "DAY_5", "DAY_6",
+                                    "DAY_7", "D_FMT", "D_T_FMT",
+                                    "ERA", "ERA_D_FMT", "ERA_D_T_FMT",
+                                    "ERA_T_FMT", "MON_1", "MON_10",
+                                    "MON_11", "MON_12", "MON_2", "MON_3",
+                                    "MON_4", "MON_5", "MON_6", "MON_7",
+                                    "MON_8", "MON_9", "NOEXPR",
+                                    "RADIXCHAR", "THOUSEP", "T_FMT",
+                                    "T_FMT_AMPM", "YESEXPR"]
+    __all__.extend(vars().keys() & _conditional_constants_names)
```

Comment:
why does this need to add in `vars().keys()`? To me it seems like this should just be adding the static hand-written list (which being hand written is known not to overlap with the default `__all__`. Do these vary by platform or configured locale in some way?